### PR TITLE
Upgrade go.mod package versions

### PR DESCRIPTION
**Pull request by VaniHaripriya**
**Description of your changes:**
Resolves [RHOAIENG-7309](https://issues.redhat.com/browse/RHOAIENG-7309)

Implemented fixes on my fork and imported in my personal snyk account and made sure CVEs are eliminated. Also updated the license CSVs.

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -143,12 +143,17 @@ require (
...
 replace (
+	github.com/cloudevents/sdk-go/v2 v2.12.0 => github.com/cloudevents/sdk-go/v2 v2.15.1
 	github.com/go-git/go-git/v5 v5.4.2 => github.com/go-git/go-git/v5 v5.11.0
 	github.com/mattn/go-sqlite3 => github.com/mattn/go-sqlite3 v1.9.0
+	github.com/mattn/go-sqlite3 v1.9.0 => github.com/mattn/go-sqlite3 v1.14.18
 	github.com/prometheus/common => github.com/prometheus/common v0.26.0
 	go.mongodb.org/mongo-driver => go.mongodb.org/mongo-driver v1.4.4
 	go.opencensus.io => go.opencensus.io v0.22.5
+	golang.org/x/net v0.17.0 => golang.org/x/net v0.23.0
+	golang.org/x/net v0.21.0 => golang.org/x/net v0.23.0
```
Comment (gmfrasca) [inline go.mod:154]: at some point we should figure out why we're requiring v0.17.0 and v0.21.0 just to replace with v0.23.0

Comment (DharmitD) [reply go.mod:154]: +1, these could be combined into one statement replacing 0.17.0 with 0.23.0.

Comment (VaniHaripriya) [reply go.mod:154]: snyk created a CVE for 0.21.0 version, when I added only one statement to replace 0.17.0 with 0.23.0.

**Review by DharmitD — COMMENTED** [review]
/lgtm

**Review by gmfrasca — APPROVED** [review]

**Pull request merged**